### PR TITLE
Improve Build Times

### PR DESCRIPTION
**Pull request by RlonRyan**
Updated the bundled Gradle wrapper to Gradle 2.14. Local testing shows
build improvements of up to 20 seconds, an approximate 25% build time
reduction. On average, the build time differences are nominal, but
several other improvements have been made to the Gradle system since
version 2.0, including buffs to IDE compatibility. Additionally, added
the option to have Gradle login to the developer's Minecraft account by
setting the variables `minecraft_username` and `minecraft_password` in
the `${userhome}/.gradle/gradle.properties` file. This is to encourage
users to develop using valid Minecraft accounts. It is important to take
care as not to set the fields in the project's local gradle.properties
file, as to avoid commiting your Minecraft account information to the
public. Finally, added .gitignore entries for NetBeans related files, to
prevent NetBeans users from committing IDE files.

### CODE REVIEW

**Comment by tterrag1098** [comment]
We test on openjdk and oraclejdk for a reason. Why change that?

**Comment by RlonRyan** [comment]
OracleJDK is the standard JDK for java. The more important difference is the difference between Java 7 and Java 8 compiles. If you would prefer, I would be more than happy change the Java 7 JDK back to OpenJDK 7.

**Comment by tterrag1098** [comment]
The server the mod is compiled on uses openjdk8. However travis does not have this available (yet) so we compromised with openjdk7. As far as the differences between different oracle jdk versions, I think it is negligible and not worth testing every PR against.

**Comment by RlonRyan** [comment]
I see... Arguably it is not worth testing against more than one JDK version, but for the sake of security I will switch back the jdk7.

**Comment by tterrag1098** [comment]
Ok, I will let @HenryLoenwind review though, since he is the one who initially set up travis.

**Comment by RlonRyan** [comment]
Sounds good! I hope these modifications will work well for you, as they have for me in the past.

File: .travis.yml [hunk 1]
```diff
@@ -1,11 +1,53 @@
...
+before_cache:
+  - rm -f $HOME/.gradle/caches/modules-2/modules-2.lock
+  - rm -f $HOME/.gradle/caches/minecraft/ForgeVersion.json
+  - rm -f $HOME/.gradle/caches/minecraft/ForgeVersion.json.etag
+  - rm -rf $HOME/.gradle/caches/minecraft/deobfedDeps
+
+# Cache libs, as to improve speed.
 cache:
   directories:
-    - '$HOME/.m2'
```
Comment (HenryLoenwind) [inline .travis.yml]: why?

File: .travis.yml [hunk 2]
```diff
@@ -1,11 +1,53 @@
+# Tell Travis which branches to build.
+branches:
```
Comment (HenryLoenwind) [inline .travis.yml]: that doesn't fit out git setup at all

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,6 +1,6 @@
-#Tue Aug 19 17:12:51 CDT 2014
```
Comment (HenryLoenwind) [inline gradle/wrapper/gradle-wrapper.properties]: this (and below) will be overwritten the next time we update Forge

File: build.gradle [hunk 4]
```diff
@@ -69,11 +69,11 @@ dependencies {
     compile ("com.pahimar.ee3:EquivalentExchange3:${ee3_version}:dev") {
         transitive = false
     }
-
```
Comment (HenryLoenwind) [inline build.gradle]: all those whitespace changes will make it a pain to merge this into the 1.9 bracnh

File: gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
Comment (RlonRyan) [inline gradle/wrapper/gradle-wrapper.properties]: Not if you update forge properly, it's only overwritten by the gradle wrapper task.

File: build.gradle (hunk 4 above)
Comment (RlonRyan) [inline build.gradle]: They can be undone (i.e. I can undo them for you), but keeping to an aggressive trailing white space removal policy can really help prevent future merge conflicts.

File: .travis.yml (hunk 2 above)
Comment (RlonRyan) [inline .travis.yml]: I should have removed that right off the bat, but as a suggestion, I can configure it to match any setup you want, if you just tell me how you would like it.

File: .travis.yml (hunk 1 above)
Comment (RlonRyan) [inline .travis.yml]: This prevents the cache from being constantly repackaged, in my experience, with the most possible amount of files preserved in the cache.

**Comment by RlonRyan** [comment]
So... What needs to be done here?

**Pull request closed without merging**